python/prepare_release.py

```diff
@@ -0,0 +1,296 @@
+                                f"ALTER EXTENSION citus UPDATE TO '{upcoming_minor_version}-1';"):
+        raise ValueError(f"{multi_extension_sql_path} does not have match for version")
+    print(f"### OK Current schema version updated on {multi_extension_sql_path} to {upcoming_minor_version}### ")
+
+
+def get_current_schema_from_citus_control(citus_control_file_path):
+    print(f"### Reading current schema version from  {citus_control_file_path}...### ")
+    current_schema_version = get_current_schema_version(citus_control_file_path)
+    if len(current_schema_version) == 0:
+        raise ValueError("Version info could not be found in citus.control file")
+    print(f"### OK Current schema version is {current_schema_version}### ")
+    return current_schema_version
+
+
+def update_version_with_upcoming_version_inconfig_py(config_py_path, upcoming_minor_version):
+    print(f"### Updating {config_py_path} file with the upcoming version {upcoming_minor_version}...### ")
+    if not replace_line_in_file(config_py_path, "^MASTER_VERSION =*",
+                                f"MASTER_VERSION = '{upcoming_minor_version}'"):
+        raise ValueError(f"{config_py_path} does not have match for version")
+    print(f"### {config_py_path} file updated with the upcoming version {upcoming_minor_version}...### ")
+
+
+def update_version_in_multi_extension_out(multi_extension_out_path, project_version):
+    print(f"### Updating {multi_extension_out_path} file with the project version {project_version}...### ")
+    if not replace_line_in_file(multi_extension_out_path, MULTI_EXT_MAJOR_SEARCH_PATTERN,
+                                f" {project_version}"):
+        raise ValueError(f"{multi_extension_out_path} does not have match for version")
+    print(f"### OK {multi_extension_out_path} file is updated with project version {project_version}.### ")
+
+
+def execute_autoconf_f():
+    print(f"### Executing autoconf -f command...### ")
+    run("autoconf -f")
+    print(f"### OK autoconf -f executed.### ")
+
+
+def update_version_in_configure_in(configure_in_path, project_version):
+    print(f"### Updating version on file {configure_in_path}...### ")
+    if not replace_line_in_file(configure_in_path, CONFIGURE_IN_SEARCH_PATTERN,
+                                f"AC_INIT([Citus], [{project_version}])"):
+        raise ValueError(f"{configure_in_path} does not have match for version")
+    print(f"### OK {configure_in_path} file is updated with project version {project_version}.### ")
+
+
+def create_branch(release_branch_name):
+    print(f"### Creating release branch with name {release_branch_name}...### ")
+    run(f'git checkout -b {release_branch_name}')
+    print(f"### OK {release_branch_name} created### ")
+
+
+def checkout_branch(branch_name, is_test):
+    print(f"### Checking out {branch_name}...### ")
+    run(f"git checkout {branch_name}")
+    if not is_test:
```

Comment:
When is_test is active we don't push branches. So there is not a remote counterpart. 
I didn't push while testing because when we push, it triggers a pipeline job which is unnecessary in this case

Reply:
I think it makes sense to not push the branch to remote repo during tests.

On the other hand, I couldn't quite understand what does that mean ?
>When is_test is active we don't push branches. So there is not a remote counterpart.

I think there is still a remote repo from which we clone the repo.
In that sense, I think we can still do `git pull` and I think we should do.

Reply:
If there is not a remote counterpart, we are getting an error as below. So in testing case, since there is not such branch we would get error if we pulled
![image](https://user-images.githubusercontent.com/1596331/119642112-0f9cd000-be23-11eb-833d-c50a13452bd4.png)